src/plugins/path/CMakeLists.txt

```diff
@@ -1,5 +1,17 @@
+include (SafeCheckSymbolExists)
 include (LibAddMacros)
 
+set (plugin path)
+
+if (DEPENDENCY_PHASE)
+	add_definitions (-D_GNU_SOURCE)
+	safe_check_symbol_exists (euidaccess "unistd.h" TEST_EUIDACCESS)
+
+	if (NOT TEST_EUIDACCESS)
+		remove_plugin (${plugin} "could not find euidaccess.")
```

Comment:
As far as I can tell it should not be too hard assuming that `HAVE_EUIDACESS` is only defined, if `euidaccess` was found (use kdbconfig.h.in for that), the following should work.

```c
static int validatePermission (Key * key, Key * parentKey)
{
#ifdef HAVE_EUIDACCESS
    // ... put actual source code here
#else
    return 1;
#endif // HAVE_EUIDACCESS
}
```
Also surround `isUserInGroup` if an `#ifdef HAVE_EUIDACCESS`. Although this would mean that `infos/metadata` is wrong.

Which is why I would suggest checking for `check/permission/types` `check/permission/user` before the `#ifdef` and in the else case I would issue a warning, if any `check/permission` metadata was found. The warning should along the lines of "Could not check permissions, because Elektra was compiled without euidaccess."

Reply:
Also I noticed the documentation for validatePermission says
> @return 0 if success or -1 for failure

but actually `1` is returned in case of success